hooks/command

```diff
@@ -1,60 +1,160 @@
+        kubectl patch deployment/${deployment_name} --namespace="${namespace}" --type="json" -p="$(cat /tmp/${BUILDKITE_COMMIT}.json)"
     fi
 
-    kubectl set image deployment/$deploymentName $containerstring --record -n $namespace
+    kubectl set image deployment/$deployment_name $container_string --record -n $namespace
 
-    kubectl annotate deployment/$deploymentName kubernetes.io/change-cause="${annotation}" -n $namespace
+    kubectl annotate deployment/$deployment_name kubernetes.io/change-cause="${annotation}" -n $namespace
 
-    if kubectl rollout status deployment/$deploymentName -n $namespace --timeout=$timeout ; then
+    if kubectl rollout status deployment/$deployment_name -n $namespace --timeout=$timeout ; then
         echo "--- :bk-status-passed: Deploy success"
     else
         if [[ "$BUILDKITE_PLUGIN_K8S_DEPLOY_ROLLBACKONFAIL" == "true" ]]; then
             echo "--- :bk-status-failed: Deploy failed. Rolling back to previous revision"
-            kubectl rollout undo deployment/$deploymentName -n $namespace
+            kubectl rollout undo deployment/$deployment_name -n $namespace
         else
             echo "--- :bk-status-failed: Deploy failed"
         fi
         exit 1
     fi
+}
+
+function promoteImage () {
+    # Login to source repository
+
+    aws ecr get-login-password --region $BUILDKITE_PLUGIN_K8S_DEPLOY_SOURCEIMAGE_REGION | docker login --username AWS --password-stdin $BUILDKITE_PLUGIN_K8S_DEPLOY_SOURCEIMAGE_REPOSITORYURL
+    if [ $? -ne 0 ]; then
+        echo "--- :bk-status-failed: Promotion failed. Authentication failed with source repository"
+        exit 1
+    fi
+
+    # Test if the image already exists on the kubernetes host
+
+    local should_cleanup_source_image=1
+
+    if docker inspect --type=image $BUILDKITE_PLUGIN_K8S_DEPLOY_SOURCEIMAGE_REPOSITORYURL/$BUILDKITE_PLUGIN_K8S_DEPLOY_SOURCEIMAGE_IMAGE ; then
+        echo "Image tag already exists on host"
+        should_cleanup_source_image=0
+    fi
+
+    # Pull the source image
+
+    docker pull $BUILDKITE_PLUGIN_K8S_DEPLOY_SOURCEIMAGE_REPOSITORYURL/$BUILDKITE_PLUGIN_K8S_DEPLOY_SOURCEIMAGE_IMAGE
+    if [ $? -ne 0 ]; then
+        echo "--- :bk-status-failed: Promotion failed. Failed to pull source image"
+        exit 1
+    fi
+
+    # Loop over destinations and push in parallel
+
+    local i=0
+    local destination_image_envvar="BUILDKITE_PLUGIN_K8S_DEPLOY_DESTINATIONIMAGES_${i}_IMAGE"
+    local destination_repository_url_envvar="BUILDKITE_PLUGIN_K8S_DEPLOY_DESTINATIONIMAGES_${i}_REPOSITORYURL"
+    local destination_region_envvar="BUILDKITE_PLUGIN_K8S_DEPLOY_DESTINATIONIMAGES_${i}_REGION"
+
+    local push_pids=()
+
+    while [[ -n "${!destination_image_envvar:-}" ]] ; do
+        # Authenticate with the destination region
+        aws ecr get-login-password --region ${!destination_region_envvar} | docker login --username AWS --password-stdin ${!destination_repository_url_envvar}
```

Comment:
Do you need `!` here?

Reply:
`destination_region_envvar` and `destination_repository_url_envvar` are both strings that contain names of env vars and not the value, you need the `!` to get the value of the env var